liquidctl/driver/asetek_pro.py

```diff
@@ -0,0 +1,240 @@
+        msg = self._end_transaction_and_read(5)
+        pump_speed = (msg[3] << 8) + msg[4]
+        status.append(('Pump speed', pump_speed, 'rpm'))
+        self._write([_PRO_CMD_READ_FIRMWARE])
+        msg = self._end_transaction_and_read(7)
+        firmware = '{}.{}.{}.{}'.format(*tuple(msg[3:7]))
+        status.append(('Firmware version', firmware, ''))
+        return status
+
+    def set_color(self, channel, mode, colors, time_per_color=1, time_off=None,
+                  alert_threshold=_HIGH_TEMPERATURE, alert_color=[255, 0, 0],
+                  speed=3, temps=['35', '45', '55'], **kwargs):
+        """Set the color mode for a specific channel."""
+        if mode not in _PRO_COLOR_CHANGE_MODES:
+            raise KeyError(f'Unknown lighting mode {mode}')
+        colors = list(colors)
+        _LOGGER.debug('color count = %d', len(colors))
+        _LOGGER.debug('color change speed %d', int(speed))
+        if mode == 'alert':
+            self._write([0x5f, int(temps[0]), 0x00, int(temps[1]), 0x00, int(temps[2]), 0x00] + colors[0] + colors[1] + colors[2])
+            self._end_transaction_and_read(6)
+            self._write([0x5E, 0x01])
+            self._end_transaction_and_read(3)
+            return
+        if mode == 'fixed':
+            colors = [colors[0], colors[0]]
+        if mode != 'rainbow':
+            self._write_color(colors)
+        if mode != 'fixed':
+            self._write_color_change_speed(mode, speed)
+        self._write(_PRO_COLOR_CHANGE_MODES[mode])
+        self._end_transaction_and_read(3)
+
+    def set_speed_profile(self, channel, profile, **kwargs):
+        """Set channel to follow a speed duty profile."""
+        channel = channel.lower()
+        if channel == 'pump':
+            raise NotSupportedByDevice()
+        if not channel.startswith('fan'):
+            raise KeyError(f'Unknown channel: {channel}')
+        adjusted = self._prepare_profile(profile, 0, 100, _MAX_PROFILE_POINTS)
+        for temp, duty in adjusted:
+            _LOGGER.info('setting %s PWM point: (%i°C, %i%%), device interpolated',
+                        channel, temp, duty)
+        temps, duties = map(list, zip(*adjusted))
+        # Need to write curve for each fan in channel
+        for i in self._get_fan_indexes(channel):
+            _LOGGER.info('setting speed for fan %d', i+1)
+            self._write([_PRO_CMD_WRITE_FAN_CURVE, i] + temps + duties)
+            self._end_transaction_and_read(32)
+
+    def set_fixed_speed(self, channel, duty, **kwargs):
+        """Set channel to a fixed speed duty."""
+        channel = channel.lower()
+        if channel.startswith('fan'):
+            duty = clamp(duty, 0, 100)
+            # Need to write curve for each fan in channel
+            for i in self._get_fan_indexes(channel):
+                _LOGGER.info('setting speed for fan %d to %d', i+1, duty)
+                self._write([_PRO_CMD_WRITE_FAN_SPEED, i, duty])
```

Comment:
Agreed, this will probably help.

@cluxter, can you check that *all* fan channels are set to sensible speeds after each step of the following sequence?

```
set fan speed 100
set fan speed 0 0 50 100
set fan speed 100
set fan1 speed 0 0 50 100
set fan2 speed 20
```

Reply:
They all work very well!!
I also set this:
```
set fan speed 30 0 40 100
```
and ran a full kernel compilation. Fans got faster and faster up to their max when temperature went up to 40°C (and more, since the pump was in "quiet" mode), and slower and slower when I stopped the compilation and the temperature was declining, down to ~200 RPM when temperature reached 30.4°C again.

Also, now the fans start almost instantaneously! I tested going from speed 0 to speed 100, and it takes up to 3 seconds for the fans to be started, which is pretty good in my opinion (I have no idea if we could even improve this?). But once they are started, they reach their speed very very quickly, about 2 seconds if not less. It's awesome!
I noticed that they don't necessarily start at the same time, so maybe this could be improved too, even if this is good enough for the moment. As you can see, I like optimized stuff, so this is something I would enjoy working on once all the features are fully implemented (since I know Python).

I tried to set the pump mode on something other than "quiet" but didn't succeed. From what I see from the modifications, it looks intentional for the moment, so I assume you'll implement it later on.

I also tried:
```
set ring color fixed ff0000
set ring color fixed 00ff00
```
and it worked.

However:
```
set fan color fixed ff0000
```
or
```
set fan1 color fixed ff0000
```
didn't work, without any error though. I think it's my setup, I will check if I'm at least able to change the fan colors on Windows and let you know.

Reply:
The only RGB the Corsair PRO comes with is on the pump. If you've added your own RGB fans, I'm not sure if the driver should control them. 
I've changed the set color function so the only channel that works is now `pump` to make this clear.

Reply:
Is there an RGB header on the pump for non-stock RGB fans?

Reply:
No, the pump has 2 or 3 (depending on model) 4-pin fan headers. However it doesn't have any RGB headers.